python-checks/src/main/java/org/sonar/python/checks/tests/AssertOnTupleLiteralCheck.java

```diff
@@ -43,12 +43,18 @@ public void initialize(Context context) {
 
         var issue = ctx.addIssue(tuple, MESSAGE);
 
-        if (tuple.leftParenthesis() != null && tuple.rightParenthesis() != null) {
+        if (tuple.leftParenthesis() != null && tuple.rightParenthesis() != null && !tuple.elements().isEmpty()) {
           // defensive condition
-          issue.addQuickFix(PythonQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
-            .addTextEdit(TextEditUtils.remove(tuple.leftParenthesis()))
-            .addTextEdit(TextEditUtils.remove(tuple.rightParenthesis()))
-            .build());
+          var quickfixBuilder = PythonQuickFix.newQuickFix(QUICK_FIX_MESSAGE)
+            .addTextEdit(TextEditUtils.remove(tuple.leftParenthesis()));
+
+          if(tuple.elements().size() == 1 && !tuple.commas().isEmpty()) {
+            quickfixBuilder.addTextEdit(TextEditUtils.replaceRange(tuple.commas().get(0), tuple.rightParenthesis(), ""));
+          } else {
+            quickfixBuilder.addTextEdit(TextEditUtils.remove(tuple.rightParenthesis()));
+          }
```

Comment:
I am not sure this would work with `assert (a,b,c,)` as this only check for single element tuples. Is that on purpose?

Reply:
Oh, ups. No it's not intentional. I wasn't aware that trailing commas are allowed in general :sweat:

Reply:
Now that I'm thinking about this, I'm not sure if the quick fix actually makes sense. For an assert like `assert (a,b)` there are two ways to correct this with different semantics. it could be `assert a, b` (where `b` is the error message), or it could be `assert a and b`. The current quick fix assumes that it is the first one. 
However, given something like `assert (a,b,c)` the only sensible way to correct that is to generate `assert a and b and c`. But the current quick fix would generate `assert a, b, c` which isn't syntactically correct.

Reply:
@joke1196, what do you think? Would it make more sense to always join the different elements with `and`?

Reply:
So as discussed we should provide the quick fix only for the single element tuple, as in other cases we risk breaking the code.